### PR TITLE
Dovecot replication v2

### LINKED ISSUE #39: Setup dovecot peer replication
In French: https://www.swordarmor.fr/replication-mastermaster-entre-deux-serveurs-mail-avec-dovecot-dsync.html
Official documentation : https://wiki.dovecot.org/Replication

### PR DESCRIPTION
**Pull request by hugoShaka**
### Description
Added dovecot replication over tcp.
TLS not yet supported.

Fixes #39

### Checklist
- [x] Molecule tests are passing
- [ ] Extended the README / documentation, if necessary
- [x] Test for the feature added

### CODE REVIEW

File: roles/ispmail-dovecot/tasks/main.yml [hunk 1]
```diff
@@ -63,41 +63,45 @@
...
+    - 10-ssl.conf
+    - auth-sql.conf.ext
 
-- name: Copy service configuration (10-master.conf)
+- name: Copy static configuration files
   copy:
-    src: 10-master.conf
-    dest: /etc/dovecot/conf.d/10-master.conf
+    src: "{{ item }}"
+    dest: "/etc/dovecot/conf.d/{{ item }}"
```
Comment (HadrienPatte) [inline roles/ispmail-dovecot/tasks/main.yml]: The quotes are not neccesary there.

File: roles/ispmail-dovecot/tasks/main.yml [hunk 2]
```diff
@@ -63,41 +63,45 @@
...
   notify: restart dovecot
+  loop:
+    - 10-master.conf
+    - 15-mailboxes.conf
+    - 20-lmtp.conf
+    - 90-sieve.conf
 
-- name: Copy SSL configuration (10-ssl.conf)
+- name: Copying replication conf (if a peer is defined)
```
Comment (HadrienPatte) [inline roles/ispmail-dovecot/tasks/main.yml]:
```suggestion
- name: Copy replication conf (if a peer is defined)
```

File: roles/ispmail-dovecot/tasks/main.yml [hunk 3]
```diff
@@ -63,41 +63,45 @@
...
+    - 90-plugin.conf
+  when: ispmail_dovecot_replication_peer is defined
   notify: restart dovecot
 
-- name: Copy mailboxes configuration (15-mailboxes.conf - autocreate folders)
-  copy:
-    src: 15-mailboxes.conf
-    dest: /etc/dovecot/conf.d/15-mailboxes.conf
-  notify: restart dovecot
+- name: Removing replication conf (if no peer is defined)
```
Comment (HadrienPatte) [inline roles/ispmail-dovecot/tasks/main.yml]:
```suggestion
- name: Remove replication conf (if no peer is defined)
```

File: roles/ispmail-dovecot/templates/10-mail.conf.j2 [hunk 4]
```diff
@@ -209,7 +209,8 @@ namespace inbox {
 
 # Space separated list of plugins to load for all services. Plugins specific to
 # IMAP, LDA, etc. are added to this list in their own .conf files.
-#mail_plugins =
+mail_plugins = $mail_plugins
+{%- if ispmail_dovecot_replication_peer is defined %} notify replication {%endif%}
```
Comment (HadrienPatte) [inline roles/ispmail-dovecot/templates/10-mail.conf.j2]:
```suggestion
{% if ispmail_dovecot_replication_peer is defined %}notify replication{% endif %}
```

Comment (hugoShaka) [reply roles/ispmail-dovecot/templates/10-mail.conf.j2]: This dash is intended ;)

File: roles/ispmail-dovecot/tasks/main.yml (hunk 1 above)
Comment (hugoShaka) [reply roles/ispmail-dovecot/tasks/main.yml]: I don't really get it, what would you want to do ?

Comment (HadrienPatte) [reply roles/ispmail-dovecot/tasks/main.yml]: You could just write
```suggestion
    dest: /etc/dovecot/conf.d/{{ item }}
```
But this is purely aesthetic

Comment (hugoShaka) [reply roles/ispmail-dovecot/tasks/main.yml]: Ugh, what the hell ? Do you have documentation about that ?

Comment (HadrienPatte) [reply roles/ispmail-dovecot/tasks/main.yml]: If the value in a key/value mapping starts with a `{`, YAML assumes it to be the begining of an  inline array, so we must quote values that begin with a variable enclosed in `{{` for it to be interpreted correctly.
But if there is any other caracter prior to the variable, the quotes are no longer necessary, as YAML won't try to interpret the `{` as the begining of an array.
So this is ok:
```yaml
src: files/{{ inventory_hostname }}
```
But this is not (and should be quoted):
```yaml
dest: {{ install_path }}/Build
```

Comment (HadrienPatte) [reply roles/ispmail-dovecot/tasks/main.yml]: The only related documentation I could find is [this](https://yaml.org/spec/1.2/spec.html#}%20end%20flow%20mapping//) but may not be very helpful..

Comment (hugoShaka) [reply roles/ispmail-dovecot/tasks/main.yml]: Wow, thanks. I thought substitution was only done in double quoted strings. All of this makes more sense now !

**Pull request merged**